coi/src/document.rs

```diff
@@ -22,7 +22,7 @@ pub trait Document {
     fn id(&self) -> &Self::Id;
 
     /// Gets the embedding of the document.
-    fn embedding(&self) -> &NormalizedEmbedding;
+    fn embeddings<'a>(&'a self) -> Box<dyn Iterator<Item = &'a NormalizedEmbedding> + 'a>;
```

Comment:
it's a trait, using `impl Trait` in traits isn't supported without using hacks which hinder rust-analyzer and internally produce the same as the code above just with more complexity.

IMHO we should in general use `dyn` more where usable as it can make code more compact, faster to build and any performance drawback of (reliable branch predicted) vtable calls _tends_ to be armortized by e.g. more cache friendly code